src/App.js

```diff
@@ -1,196 +1,211 @@
 import React, { Component } from 'react';
-import './App.css';
 import axios from 'axios';
-
+const URL = 'http://13.113.246.46:8081/todos';
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
       todoList: [],
-      isAdd: false,
+      canAdd: false,
       title: '',
       content: '',
       userName: 'yuni',
-      check: false,
-      isUpdate: false,
+      isChecked: false,
+      canUpdate: false,
       id: 0,
     };
     this.handleChange = this.handleChange.bind(this);
     this.toggleAdd = this.toggleAdd.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
     this.deleteTodo = this.deleteTodo.bind(this);
-    this.preUpdateTodo = this.preUpdateTodo.bind(this);
+    this.handleTodoEdit = this.handleTodoEdit.bind(this);
     this.handleSubmitUpdate = this.handleSubmitUpdate.bind(this);
     this.handleInputChange = this.handleInputChange.bind(this);
   };
 
   componentDidMount() {
-    axios.get('http://13.113.246.46:8081/todos/userName/yuni')
+    axios.get(`${URL}/userName/yuni`)
       .then(result => {
         this.setState({
           todoList: result.data.result,
         })
       })
-      .catch(error => console.log(error));
+      .catch(error => console.error(error));
   };
 
-  componentDidUpdate() {
-    axios.get('http://13.113.246.46:8081/todos/userName/yuni')
-      .then(result => {
-        if(JSON.stringify(this.state.todoList) !== JSON.stringify(result.data.result)) {
-          this.setState({
-            todoList: result.data.result,
-            isAdd: false,
-            isUpdate: false,
-            title: '',
-            content: '',
-            userName: 'yuni',
-            check: 0,
-          })
-        }
-      })
-      .catch(error => console.log(error));
-  };
+  // componentDidUpdate(prevProps, prevState) {
+  //   const { todos: prevTodos } = prevState
+  //   const { todos } = this.state
+
+  //   if (todos !== prevTodos) {
+  //     axios.get(`${URL}/userName/yuni`)
+  //       .then(result => {
+  //         if (JSON.stringify(this.state.todoList) !== JSON.stringify(result.data.result)) {
+  //           this.setState({
+  //             todoList: result.data.result,
+  //             canAdd: false,
+  //             canUpdate: false,
+  //             title: '',
+  //             content: '',
+  //             userName: 'yuni',
+  //             isChecked: 0,
+  //           })
+  //         }
+  //       })
+  //       .catch(error => console.error(error));
+  //   }
+  // };
 
   handleChange(event) {
-    const target = event.target;
+    const { target } = event;
```

Comment:
`handleChange` 내에서 어짜피 `target.property` 형식으로 쓰이고 있는데, 그럴거면 한 빙에 `event.target`을 디스트럭쳐링하는게 낫지 않을까? 64행에서도 `target.name`을 `name`에 담던데, 61행에서 한 방에 디스트럭쳐링하면 62행 ~ 64행도 더 줄일 수 있을 듯
```suggestion
    const { type, checked, value, name } = event.target;
```

Reply:
const value = target.type === 'checkbox' ? target.checked : target.value;
여기서 좀 꼬이지 않을까요

Reply:
`value` 변수에 값을 할당하는 62행을 66행과 합치면 되지 않을까? 아니면 변수명을 아예 다르게 지어도 좋고~

```javascript
this.setState({
  [name]: type === 'checkbox' ? checked : value,
});
```